src/handshake.rs

```diff
@@ -107,6 +107,32 @@ impl Handshake {
             }
         }))
     }
+
+    /// Get a forwarded IP address of the remote connection.
+    ///
+    /// This is the preferred method of obtaining the client's trusted IP address.
+    /// It will attempt to retrieve the forwarded IP address for `trusted` depth
+    /// of reverse-proxies based on request headers, `trusted=0` returns the direct
+    /// address of the peer.
+    ///
+    /// # Note
+    /// This assumes that the peer is a client. If you are implementing a
+    /// WebSocket client and want to obtain the address of the server, use
+    /// `Handshake::peer_addr` instead.
+    ///
+    /// This method does not ensure that the address is a valid IP address.
+    #[allow(dead_code)]
+    fn forwarded_addr(&self, trusted: usize) -> Result<Option<String>> {
```

Comment:
This function would be helpful, if it was accessible.

Github stops me from suggesting, so I just write it out.
The `#[allow(dead_code)]` is not nessesary when the function gets a `pub`

Reply:
This is modelled exactly along the existing functions and API. I'd rather have this merged with no fuss than propose an new API.